File: poc-1-1.py
import duckdb
import os
import time

conn = duckdb.connect()

st = time.time()

for i in range(80_000):
    fileNum = i + 1
    conn.execute(f"CREATE TABLE etp_transform_{fileNum} AS SELECT * FROM read_csv_auto('./generatedFiles/ETP-Generated-File-{fileNum}.csv')")
    conn.execute(f"UPDATE etp_transform_{fileNum} SET Name = 'TransformedName'")
    conn.execute(f"COPY (SELECT * FROM etp_transform_{fileNum}) TO './exportedFiles-poc1.1.1/ETP-Transformed-File-{fileNum}.csv' (HEADER, DELIMITER ',')")
    conn.execute(f"DROP TABLE etp_transform_{fileNum}")

et = time.time()

elapsed_time = et - st
print('Execution time:', elapsed_time, 'seconds')

File: poc-1-1.js
import duckdb from "duckdb";
import fs from "fs";

const db = new duckdb.Database(":memory:");
const baseDir = "./generatedFiles";

const fileAmount = fs.readdirSync(baseDir);

let i = 1,
  len = fileAmount.length;

console.time();

const transformation = new Promise((resolve, reject) => {
  while (i < len) {
    db.run(
      `CREATE TABLE etp_transform_${i} AS SELECT * FROM read_csv_auto('./generatedFiles/ETP-Generated-File-${i}.csv')`
    );
    db.run(`UPDATE etp_transform_${i} SET Name = 'TransformedName'`);
    db.run(
      `COPY (SELECT * FROM etp_transform_${i}) TO './exportedFiles-poc1.1/ETP-Transformed-File-${i}.csv' (HEADER, DELIMITER ',')`
    );
    db.run(`DROP TABLE etp_transform_${i}`);

    if (i === len - 1) {
      resolve();
    }

    i++;
  }
});

transformation.then(() => {
  db.all("SHOW TABLES", (err, res) => console.timeEnd());
});
